Dockerfile

```diff
@@ -0,0 +1,37 @@
+FROM python:3.11-slim-bullseye
+
+ENV DEBIAN_FRONTEND noninteractive
+ENV TERM linux
+
+# Install distribution packages, with caching.
+# `git` is needed for `versioningit`.
+RUN rm -f /etc/apt/apt.conf.d/docker-clean; echo 'Binary::apt::APT::Keep-Downloaded-Packages "true";' > /etc/apt/apt.conf.d/keep-cache
+RUN \
+    --mount=type=cache,id=apt,sharing=locked,target=/var/cache/apt \
+    --mount=type=cache,id=apt,sharing=locked,target=/var/lib/apt \
+    true \
+    && apt-get update \
+    && apt-get install --no-install-recommends --no-install-suggests --yes git ffmpeg firefox-esr xauth xvfb
+
+# Copy sources
+COPY . /src
```

Comment:
This COPY makes all commands below re-run if anything in project is changed (pip install, apt-get remove, ADD tini).

I improved on this in 41ab012

Reply:
Re-running if something has changed is intentional, no?

Reply:
Yes but changes in python should not re-run pip etc, only if setup.py (or README.rst) were changed.

Reply:
Also added to GH-25, thanks.

Reply:
Also added to GH-25, thanks.